internal/source/git.go

```diff
@@ -2,145 +2,91 @@ package source
 
 import (
 	"bytes"
-	"compress/gzip"
 	"context"
-	"encoding/json"
 	"fmt"
 	"io"
 	"io/fs"
-	"strings"
+	"os"
+	"path/filepath"
 
-	"github.com/nlepage/go-tarfs"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
 
 	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
-	"github.com/operator-framework/rukpak/internal/git"
-	"github.com/operator-framework/rukpak/internal/util"
 )
 
 type Git struct {
-	Client          client.Client
-	KubeClient      kubernetes.Interface
-	ProvisionerName string
-	PodNamespace    string
-	UnpackImage     string
-	GitClientImage  string
 }
 
-const gitBundleUnpackContainerName = "bundle"
-
-func (g *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
 	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeGit {
 		return nil, fmt.Errorf("bundle source type %q not supported", bundle.Spec.Source.Type)
 	}
 	if bundle.Spec.Source.Git == nil {
 		return nil, fmt.Errorf("bundle source git configuration is unset")
 	}
-
-	pod := &corev1.Pod{}
-	op, err := g.ensureUnpackPod(ctx, bundle, pod)
-	if err != nil {
-		return nil, err
-	}
-	if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated || pod.DeletionTimestamp != nil {
-		return &Result{State: StatePending}, nil
+	gitsource := bundle.Spec.Source.Git
+	if gitsource.Repository == "" {
+		// This should never happen because the validation webhook rejects git bundles without repository
+		return nil, fmt.Errorf("missing git information")
 	}
 
-	switch phase := pod.Status.Phase; phase {
-	case corev1.PodPending:
-		return pendingGitPodResult(pod), nil
-	case corev1.PodRunning:
-		return &Result{State: StateUnpacking}, nil
-	case corev1.PodFailed:
-		return nil, g.failedPodResult(ctx, pod)
-	case corev1.PodSucceeded:
-		return g.succeededPodResult(ctx, bundle, pod)
-	default:
-		return nil, g.handleUnexpectedPod(ctx, pod)
+	// Set options for clone
+	cloneOpts := git.CloneOptions{
+		URL:      gitsource.Repository,
+		Progress: &bytes.Buffer{},
```

Comment:
Is this the setting for where the git progress logs should be sent to? Maybe we could pipe them to stdout so users can see those logs? It would be easier for users to diagnose errors that way, at the expense of some potentially noisy logs.

Reply:
OK.  I'll put the progress into the log.  It's usually a few lines and it doesn't look too bad.  Thanks!